### PR TITLE
prevent potential panic: return immediately if there exists error

### PR DESCRIPTION
**Pull request by maipbui**
#### Why I did it
Variable `v` is likely modified and later used on error. In some cases this could result in panics due to a nil dereference
#### How I did it
return immediately if there exists error
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

**Comment by ganglyu** [comment]
Please add unit test to cover this change.

File: sonic_data_client/non_db_client.go [hunk 1]
```diff
@@ -566,6 +566,7 @@ func (c *NonDbClient) Get(w *sync.WaitGroup) ([]*spb.Value, error) {
 		v, err := getter()
 		if err != nil {
 			log.V(3).Infof("PollRun getter error %v for %v", err, v)
+			return nil, err
```
Comment (ayelrod) [inline sonic_data_client/non_db_client.go:569]: Would it make more sense here to `continue` instead of `return`?

Comment (qiluo-msft) [reply sonic_data_client/non_db_client.go:569]: I think caller could determine whether retry based on return value of this function.

**Review by qiluo-msft — DISMISSED** [review]

File: sonic_data_client/client_test.go [hunk 2]
```diff
@@ -322,3 +324,28 @@ func TestParseTarget(t *testing.T) {
...
+	}
+
+	// Create a NonDbClient with the mocked dataGetFunc
+	client := &NonDbClient{
+		path2Getter: path2Getter,
+	}
+
+	var w *sync.WaitGroup
+	_, err := client.Get(w)
+	if err == nil {
```
Comment (ganglyu) [inline sonic_data_client/client_test.go:348]: You can compare the error here.

**Review by ganglyu — APPROVED** [review]

**Pull request merged**